src/KeyboardInputModal.js

```diff
@@ -0,0 +1,216 @@
+            const optionText = this.props.intl.formatMessage({ id: messageId });
+            selectOptionElements.push(<option key={schemeName} value={schemeName}>
+                {optionText}
+            </option>);
+        })
+
+        return (<select
+            className="KeyboardInputModal__content__schemeDropdown"
+            defaultValue={this.props.keyboardInputSchemeName}
+            disabled={!this.props.keyBindingsEnabled}
+            onChange={this.props.onChangeKeyboardInputScheme}>
+            {selectOptionElements}
+        </select>);
+    }
+
+    render () {
+        return(
+            <Modal
+                onHide={this.props.onHide}
+                show={this.props.show}
+                aria-modal={true}
+                role="dialog"
+                dialogClassName='KeyboardInputModal'
+            >
+                <Modal.Body className='KeyboardInputModal__content'>
+                    <h2 className="KeyboardInputModal__content__title">
+                        <KeyboardIcon/>
+                        <div>
+                            <FormattedMessage id='KeyboardInputModal.Title'/>
+                        </div>
+                    </h2>
+
+                    <div className="KeyboardInputModal__content__toggleBar">
+                        <div className="KeyboardInputModal__content__toggleBar__label">
+                            <FormattedMessage id='KeyboardInputModal.Toggle.Label'/>
+                        </div>
+                        <div className="KeyboardInputModal__content__toggleBar__toggle">
+                            <div>
+                                <FormattedMessage id='KeyboardInputModal.Toggle.Off'/>
+                            </div>
+                            <ToggleSwitch
+                                ariaLabel={this.props.intl.formatMessage({id: "KeyboardInputModal.Toggle.AriaLabel"})}
+                                className="KeyboardInputModal__content__toggle"
+                                contentsTrue=""
+                                contentsFalse=""
+                                value={this.props.keyBindingsEnabled}
+                                onChange={this.props.onChangeKeyBindingsEnabled}
+                            />
+                            <div>
+                                <FormattedMessage id='KeyboardInputModal.Toggle.On'/>
+                            </div>
+                        </div>
+
+
+
+
+                    </div>
+
+                    {this.renderKeyboardSchemeMenu()}
+
+                    <ul className={"KeyboardInputModal__content__list" + (this.props.keyBindingsEnabled ? "": " KeyboardInputModal__content__list--disabled")}>
+                        {this.renderKeyBindings()}
+                    </ul>
+
+                    <div className="KeyboardInputModal__content__footer">
+                        <Button className="KeyboardInputModal__content__cancelButton" onClick={this.props.onHide}>
```

Comment:
Both "Cancel" and "Done" buttons are functioning the same on click, could you confirm if that's intended with Sepideh please? If they are the same, I am not sure why we are using two buttons for that.

Reply:
I'll ask her, I had the same thought but wanted to start with what she'd designed.

Reply:
I talked to Sepi, she wants the "Done" button to actually make the state changes, and "Cancel" to abandon them.  This will require a light refactor but isn't difficult.

Reply:
I rearranged things so that:

1. Changes are displayed onscreen as they're made.
2. Changes are passed on to the app when "Done" is hit.
3. Changes are reset when "Cancel" is hit or the dialogue is closed without hitting "Done".